ruby/reschedule_or_cancel_unfilled_service.rb

```diff
@@ -0,0 +1,216 @@
+# Service for handling the automatic rescheduling/canceling of unfilled bookings.
+module RescheduleOrCancelUnfilledService
+  MAX_OFFSET = 8 * 7 # up to 8 weeks away
+
+  class << self
+    def auto_rbu_or_cbu_candidate?(booking, date_start: booking.date_start)
+      return false unless booking.confirmed
+      return false if booking.provider.present?
+
+      window = auto_rbu_or_cbu_minutes(region: booking.region, service: booking.service).minutes
+
+      ((date_start - window)...date_start).cover?(Time.current)
+    end
+
+    # Get bookings in need of auto-RBU/CBU
+    # @return array of eligible bookings
+    def bookings_to_reschedule_or_cancel
+      regions  = Region.select { |region| ConfigParam.regional_auto_rbu?(region) }
```

Comment:
We could explore the option of using `.find_each` and measure/benchmark if there is a performance difference.